actix-session/src/storage/redis_rs.rs

```diff
@@ -83,31 +101,53 @@ impl RedisSessionStore {
     /// A fluent API to configure [`RedisSessionStore`].
     /// It takes as input the only required input to create a new instance of [`RedisSessionStore`] - a
     /// connection string for Redis.
+    #[cfg(feature = "redis-session")]
     pub fn builder<S: Into<String>>(connection_string: S) -> RedisSessionStoreBuilder {
         RedisSessionStoreBuilder {
             configuration: CacheConfiguration::default(),
             connection_string: connection_string.into(),
         }
     }
 
+    /// A fluent API to configure [`RedisSessionStore`].
+    /// It takes as input the only required input to create a new instance of [`RedisSessionStore`] - a
+    /// pool object for Redis.
+    #[cfg(not(feature = "redis-session"))]
+    pub fn builder<P: Into<Pool>>(pool: P) -> RedisSessionStoreBuilder {
+        RedisSessionStoreBuilder {
+            configuration: CacheConfiguration::default(),
+            pool: pool.into(),
+        }
+    }
+
     /// Create a new instance of [`RedisSessionStore`] using the default configuration.
     /// It takes as input the only required input to create a new instance of [`RedisSessionStore`] - a
     /// connection string for Redis.
-    pub async fn new<S: Into<String>>(
-        connection_string: S,
-    ) -> Result<RedisSessionStore, anyhow::Error> {
+    #[cfg(feature = "redis-session")]
+    pub async fn new<S: Into<String>>(connection_string: S) -> Result<RedisSessionStore, Error> {
```

Comment:
i'd rather not introduce a crate feature that makes things non-additive.

Reply:
Sorry, but I didn't understand what you meant by that, what do you mean? About the new `deadpool-redis` feature or...?